### PR TITLE
[JUJU-3849] Introduce a new ApiServerSuite fixture to replace JujuConnSuite

### PR DESCRIPTION
**Pull request by wallyworld**
This PR introduces a new ApiServerSuite test fixture which will be used in place of JujuConnSuite. Unlike the latter, which provided a kitchen sink worth of functionality - including a full dummy environ, hosted model, heavy weight api server workers, and lots of state on the fixture - this new fixture provides:

- persistence layer (mongo and [sql/dq]lite)
- http service mux (for macaroon discharge endpoints etc when needed)
- minimum api server 
- configurable api server workers / components
- minimal state held on the fixture

When instantiated, the data model has only the controller model. Any hosted models need to be created.

It also provides the One True Way for tests to stand up an api server. Previously, we were using at least 3 similar but different test helpers / fixtures. So a lot of code is now deleted. Also, some tests were moved to a better location, eg the embedded cli apiserver tests.

Also deleted are a number of tests related to storing logs in db - we are moving away from this and rather than spend time porting the tests across, they are deleted. Plus the metrics have had bash tests added in another PR, so delete the feature tests for those. And the undertaker api is also well tested in model teardown in bash tests.

The code is test changes only, expect for deleting a create model validation - we were checking the storage provider registry is not nil on creation, but it can be nil as it is checked elsewhere before use.

The biggest test changes have been done - apiserver and api top level tests. Some facade tests are also done. The remaining ones are left for another PR - it's largely mechanical at this point. So far, the dummy environ is not needed, and for the vast majority of the remaining tests, will not be.

## QA steps

Run the tests

### CODE REVIEW

**Comment by wallyworld** [comment]
Got 2 tests to fix

File: juju/testing/apiserver.go [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+	// AdminUser is the default admin user for a controller.
+	AdminUser = names.NewUserTag("admin")
+
+	// DefaultCloudRegion is the default cloud region for a controller model.
+	DefaultCloudRegion = "dummy-region"
+
+	// DefaultCloud is the default cloud for a controller model.
+	DefaultCloud = cloud.Cloud{
+		Name:      "dummy",
+		Type:      "ec2",
```
Comment (hpidcock) [inline juju/testing/apiserver.go:68]: would be good to not use a real provider name here

File: juju/testing/apiserver.go [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+			DefaultCredentialTag: defaultCredential,
+		},
+		MongoSession:  session,
+		AdminPassword: AdminSecret,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	s.controller = ctrl
+
+	// Set the api host ports in state.
+	sHsPs := []network.SpaceHostPorts{{
```
Comment (hpidcock) [inline juju/testing/apiserver.go:249]: ahhh yes of course, the magical sHsPs.

**Review by hpidcock — APPROVED** [review]

File: juju/testing/apiserver.go (hunk 1 above)
Comment (wallyworld) [reply juju/testing/apiserver.go:68]: I tried it but too much stuff still looks for a cloud called "dummy"
Will revisit once jujuconnsuite is gone

**Pull request merged**